core/sql/optimizer/OptLogRelExpr.cpp

```diff
@@ -2394,7 +2394,7 @@ Join::synthConstraints(NormWA * normWAPtr)
   (void) leftGA.hasCardConstraint(minLeft,maxLeft);
   (void) rightGA.hasCardConstraint(minRight,maxRight);
 
-  if (maxLeft == 1)
+  if (maxLeft == 1 || isIndexJoin_)
```

Comment:
What happens in the case of joining a non-unique index to a base table? Would we want to have probe caches then?

Reply:
The non-unique index contains one row for each base table row, and it contains the clustering key of that row. So, there is still a 1:1 match between the non-unique index and the base table/clustering index, and a probe cache would not have any cache hits.

Reply:
Thanks. I guess I was thinking about joins to the index from elsewhere. But that's not what this code is dealing with.